pandas/core/internals/managers.py

```diff
@@ -798,24 +812,43 @@ def as_array(self, transpose: bool = False) -> np.ndarray:
             arr = np.empty(self.shape, dtype=float)
             return arr.transpose() if transpose else arr
 
+        should_copy = copy or na_value is not lib.no_default
```

Comment:
why is this needed?

Reply:
It's to avoid making a second copy in certain cases where we can detect this (eg with multiple dtypes (going through `_interleave`), the constructed array is never the original values, so an additional copy is not needed

Reply:
then i would just make this
```
copy = copy or na_value is not lib.no_default
```

and add an explanation line